### PR TITLE
feat(worker-playground): delay preview environment setup

### PR DESCRIPTION
**Pull request by edmundhung**
## What this PR solves / how to test

Fixes [DEVX-1410](https://jira.cfdata.org/browse/DEVX-1410).

## Author has addressed the following

- Tests
  - [ ] TODO (before merge)
  - [ ] Tests included
  - [x] Tests not necessary because: Worker playground has no tests at the moment
- E2E Tests CI Job required? (Use "e2e" label or ask maintainer to run separately)
  - [ ] I don't know
  - [ ] Required
  - [x] Not required because: No impact on other projects
- Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))
  - [ ] TODO (before merge)
  - [x] Changeset included
  - [ ] Changeset not necessary because:
- Public documentation
  - [ ] TODO (before merge)
  - [ ] Cloudflare docs PR(s): 
  - [x] Documentation not necessary because: No feature change

### CODE REVIEW

File: packages/workers-playground/src/QuickEditor/HTTPTab/HTTPTab.tsx [hunk 1]
```diff
@@ -89,22 +93,33 @@ export function HTTPTab() {
...
 		<Div display="flex" flexDirection="column" width="100%">
 			<Form
 				display="flex"
-				onSubmit={(e) => void onSendRequest(e)}
+				onSubmit={(e) => {
+					e.preventDefault();
+					preview();
+
+					if (url === previewUrl) {
+						setRefreshTimestamp(new Date().toISOString());
```
Comment (edmundhung) [inline packages/workers-playground/src/QuickEditor/HTTPTab/HTTPTab.tsx:118]: The timestamp has no use but trigger the effect above (`sendRequest`) to re-run.

File: packages/workers-playground/src/QuickEditor/HTTPTab/HTTPTab.tsx [hunk 2]
```diff
@@ -89,22 +93,33 @@ export function HTTPTab() {
 					setIsLoading(false);
 				}
 			}
-		},
-		[previewHash, setPreviewUrl, previewUrl, method, headers, hasBody, body]
-	);
-	const ensureDevtoolsConnected = useRef(false);
```
Comment (penalosa) [inline packages/workers-playground/src/QuickEditor/HTTPTab/HTTPTab.tsx:95]: Why has this been deleted?

File: packages/workers-playground/src/QuickEditor/HTTPTab/HTTPTab.tsx
Comment (penalosa) [inline packages/workers-playground/src/QuickEditor/HTTPTab/HTTPTab.tsx:1]: Why did this file need changed? It seems from a quick skim that it's doing roughly the same thing

File: packages/workers-playground/src/QuickEditor/HTTPTab/HTTPTab.tsx (hunk 2 above)
Comment (edmundhung) [reply packages/workers-playground/src/QuickEditor/HTTPTab/HTTPTab.tsx:95]: The variable `ensureDevtoolsConnected` was used to send the initial request. This is no longer needed as we have the `useEffect` to keep track of all the config and send a new request when needed.

File: packages/workers-playground/src/QuickEditor/HTTPTab/HTTPTab.tsx
Comment (edmundhung) [reply packages/workers-playground/src/QuickEditor/HTTPTab/HTTPTab.tsx:1]: We didn't share the URLBar implementation between the two tabs unlike the dashboard. Do you want me to fix that?

**Review by penalosa — APPROVED** [review]

**Pull request merged**